protocols/identify/src/handler.rs

```diff
@@ -90,36 +130,84 @@ pub struct Handler {
 
     /// The interval of `trigger_next_identify`, i.e. the recurrent delay.
     interval: Duration,
+
+    /// The public key of the local peer.
+    public_key: PublicKey,
+
+    /// Application-specific version of the protocol family used by the peer,
+    /// e.g. `ipfs/1.0.0` or `polkadot/1.0.0`.
+    protocol_version: String,
+
+    /// Name and version of the peer, similar to the `User-Agent` header in
+    /// the HTTP protocol.
+    agent_version: String,
+
+    /// Address observed by or for the remote.
+    observed_addr: Multiaddr,
+
+    /// Information provided by the `Behaviour` upon requesting.
+    behaviour_info: Option<BehaviourInfo>,
```

Comment:
Alternatively, we could initialise this in the constructor as well. I think we can assume that `new_handler` in the behaviour only gets called after it got polled at least once!

Reply:
> Instead of tracking this here, I think it would be cleaner to:
> 
> 1. Store pending substreams in a queue
> 2. Request a new `BehaviourInfo` every time a new substream comes in
> 3. Construct a `Sending` every time you receive a `BehaviourInfo` from the behaviour by popping the oldest substream from the queue in (1)
> 
> That will work off substreams in a FIFO fashion and feed them the most up-to-date information.

Thanks, makes sense to me. But used two queues instead of one with different states for a less confusing management of them ptal Thomas.